storyscript/compiler/semantics/symbols/Symbols.py

```diff
@@ -128,7 +132,31 @@ def index(self, tree, name, type_, kind):
             type_ = type_.type()
         else:
             assert isinstance(type_, BaseType)
-        new_type = symbol.type().index(type_, kind)
+
+        cur_type = symbol.type()
+        new_type = cur_type.index(type_, kind)
+        if isinstance(cur_type, ObjectType) and new_type is not None:
+            object = new_type
```

Comment:
`object` is a keyword in Python, so better use a different name here
```suggestion
            obj = new_type
```